modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/InlineIndexScanPlannerTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.calcite.planner;
+
+import java.sql.Timestamp;
+import java.util.UUID;
+import org.apache.ignite.internal.processors.query.calcite.rel.AbstractIndexScan;
+import org.apache.ignite.internal.processors.query.calcite.schema.IgniteSchema;
+import org.apache.ignite.internal.processors.query.calcite.trait.IgniteDistributions;
+import org.junit.Test;
+
+/**
+ * Planner test for index inline scan.
+ */
+public class InlineIndexScanPlannerTest extends AbstractPlannerTest {
+    /** */
+    @Test
+    public void testInlinScan() throws Exception {
+        TestTable tbl = createTable("TBL", 100, IgniteDistributions.single(),
+            "I0", UUID.class,
+            "I1", Integer.class,
+            "I2", Long.class,
+            "I3", String.class,
+            "I4", Timestamp.class,
+            "I5", Byte.class,
+            "I6", Object.class
+        );
+
+        tbl.addIndex("IDX1", 0, 2, 4);
+        tbl.addIndex("IDX2", 2, 0);
+        tbl.addIndex("IDX3", 5, 3);
+        tbl.addIndex("IDX4", 5, 6);
+
+        IgniteSchema publicSchema = createSchema(tbl);
+
```

Comment:
`        assertPlan("SELECT I0 FROM TBL", publicSchema, isIndexScan("TBL", "IDX1")
            .and(AbstractIndexScan::isInlineScan));`
I expect that IDX1 would be used in such a case, but test failed.
Unexpected index name [exp=IDX1, act=IDX2]

Reply:
can you additionally append tests for :
tbl.addIndex(RelCollations.of(TraitUtils.createFieldCollation(1, false),
    TraitUtils.createFieldCollation(2, true)), "IDX_C1");

tbl.addIndex(RelCollations.of(TraitUtils.createFieldCollation(2, false),
    TraitUtils.createFieldCollation(1, true)), "IDX_C2");
and appropriate: 
SELECT I1 FROM TBL WHERE I1[<|>]1 and I2[<|>]5

Reply:
> I expect that IDX1 would be used in such a case, but test failed.
IDX2 should be used, because it has less columns. It doesn't matter how far required key in index from the beginning, we still need to read all inlined key fields, so index with the less total keys is prefered.

Reply:
> can you additionally append tests

I didn't get, what such a test will show? It's possible to use both indexes, there is no clues for optimizer which is better (at least without statistics), and optimizer can use random.